pandas/core/common.py

```diff
@@ -909,16 +909,17 @@ def diff(arr, n, axis=0):
         analagoust to s-s.shift(n) """
 
     n = int(n)
-    na = np.nan
     dtype = arr.dtype
+    if issubclass(dtype.type, np.integer):
```

Comment:
How does this relate to what you did below? Either way, can you test the changes for `diff` more directly? This probably should also get a bug fix line.